src/lib/y2network/sysconfig/connection_config_readers/base.rb

```diff
@@ -62,6 +62,8 @@ def connection_config
         # @return [Class]
         def connection_class
           class_name = self.class.to_s.split("::").last
+          file_name = class_name.gsub(/(\w)([A-Z])/, "\\1_\\2").downcase
```

Comment:
NP: instead of this magic you can load a type from class_name and use its methods for file_name and class_name